src/miner.cpp

```diff
@@ -374,38 +429,55 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn,  unsigned int nBlo
         TxPriorityCompare comparer(fSortedByFee);
         std::make_heap(vecPriority.begin(), vecPriority.end(), comparer);
 
+
+        // considering certs having a higher priority than any possible tx.
+        // An algorithm for managing tx/cert priorities could be devised
```

Comment:
AcceptToMemoryPool seems to accept transactions sending coins to a sidechain that is declared in a Tx that is still in the mempool. Block creation seems to not keep track of such a situation by prioritizing txs inclusion considering "dependers" and orphans. This could probably lead to a situation where an invalid block is created by omitting or including in the wrong order a sc creation transaction used by a ft transaction. This will lead CreateNewBlock to keep failing.

Reply:
seems, it has been fixed in current revision.

Reply:
Done